connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/connect/client/util/RemoteSparkSession.scala

```diff
@@ -217,4 +222,17 @@ trait RemoteSparkSession extends ConnectFunSuite with BeforeAndAfterAll {
     spark = null
     super.afterAll()
   }
+
+  /**
+   * SPARK-44259: override test function to skip `RemoteSparkSession-based` tests as default,
+   * we should delete this function after SPARK-44121 is completed.
+   */
+  override protected def test(testName: String, testTags: Tag*)(testFun: => Any)
```

Comment:
@dongjoon-hyun @HyukjinKwon By using the override `test` function, all `RemoteSparkSession-based` tests for Java 21 are ignored by default, so there is no need to add the assume condition one by one

Reply:
Does this seem less intrusive